Properly check separate flags when scanning object and building track list, fixes #3861

Gymnasiast opened the pull request.

janisozaur commented:
I don't think the #3861 is the number you wanted, is it?

Can you make the commit message shorter?

janisozaur commented:
And it sounds like the issue it fixes has been around for as while, so add it to changelog.

Gymnasiast commented:
I fixed the issue number and added it to the changelog. I could not shorten the commit message without losing important information, so I haven't done that.
